pkg/sql/table.go

```diff
@@ -268,7 +268,6 @@ type TableCollection struct {
 // of a transaction retry.
 func (tc *TableCollection) resetForTxnRetry(ctx context.Context, txn *client.Txn) {
 	if tc.timestamp != (hlc.Timestamp{}) && tc.timestamp != txn.OrigTimestamp() {
-		txn.ResetDeadline()
 		tc.releaseTables(ctx)
```

Comment:
Can you add a TODO to remove tc.timestamp . I think we should have the txn retry always call releaseTables